cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -346,8 +349,11 @@ public void updateApp(HttpRequest request, HttpResponder responder,
       throw new BadRequestException("Request body is invalid json: " + e.getMessage());
     }
 
+    boolean updateSchedules =
+      appRequest.canUpdateSchedules() == null ? appUpdateSchedules : appRequest.canUpdateSchedules();
+
     try {
-      applicationLifecycleService.updateApp(appId, appRequest, createProgramTerminator());
+      applicationLifecycleService.updateApp(appId, appRequest, createProgramTerminator(), updateSchedules);
```

Comment:
Why does the AppLifecycleHttpHandler need appUpdateSchedules property to be read. In the current code I see it is just passed to the lifecycleService. If you move it to service then you don't need it in handler so there is no duplication as such.